Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -161,97 +163,79 @@ public void Serialize<T>(Stream stream, T item)
-            if (item is int)
+            else if (item is int)
             {
                 new IonIntSerializer().Serialize(writer, Convert.ToInt32(item));
-                return;
             }
-
-            if (item is long)
+            else if (item is long)
             {
                 new IonLongSerializer().Serialize(writer, Convert.ToInt64(item));
-                return;
             }
-
-            if (item is float)
+            else if (item is float)
             {
                 new IonFloatSerializer().Serialize(writer, Convert.ToSingle(item));
-                return;
             }
-
-            if (item is double)
+            else if (item is double)
             {
                 new IonDoubleSerializer().Serialize(writer, Convert.ToDouble(item));
-                return;
             }
-
-            if (item is decimal)
+            else if (item is decimal)
             {
                 new IonDecimalSerializer().Serialize(writer, Convert.ToDecimal(item));
-                return;
             }
-
-            if (item is BigDecimal)
+            else if (item is BigDecimal)
             {
                 new IonBigDecimalSerializer().Serialize(writer, (BigDecimal)(object)item);
-                return;
             }
-
-            if (item is byte[])
+            else if (item is byte[])
             {
                 new IonByteArraySerializer().Serialize(writer, (byte[])(object)item);
-                return;
             }
-
-            if (item is string)
+            else if (item is string)
             {
                 new IonStringSerializer().Serialize(writer, item as string);
-                return;
             }
-
-            if (item is SymbolToken)
+            else if (item is SymbolToken)
             {
                 new IonSymbolSerializer().Serialize(writer, (SymbolToken)(object)item);
-                return;
             }
-
-            if (item is DateTime)
+            else if (item is DateTime)
             {
                 new IonDateTimeSerializer().Serialize(writer, (DateTime)(object)item);
-                return;
             }
-
-            if (item is System.Collections.IList) 
+            else if (item is System.Collections.IList) 
             {
                 NewIonListSerializer(item.GetType()).Serialize(writer, (System.Collections.IList)(object)item);
-                return;
             }
-
-            if (item is Guid) 
+            else if (item is Guid) 
             {
                 new IonGuidSerializer(options).Serialize(writer, (Guid)(object)item);
-                return;
             }
-
-            if (item is object) 
+            else if (item is object) 
             {
                 new IonObjectSerializer(this, options, item.GetType()).Serialize(writer, item);
-                return;
+            }
+            else
+            {
+                throw new NotSupportedException("Do not know how to serialize type " + typeof(T));

```

Comment:
Maybe a better string here is "Typeof(T) is not supported for serialization."

Reply:
Done.